src/main/java/org/broadinstitute/dsm/cf/TestBostonKitTrackerDispatcher.java

```diff
@@ -0,0 +1,145 @@
+            List<DDPInstance> ddpInstanceList = getDDPInstanceListWithRole(conn, "ups_tracking");
+            for (DDPInstance ddpInstance : ddpInstanceList) {
+                if (ddpInstance != null && ddpInstance.isHasRole()) {
+                    logger.info("tracking ups ids for " + ddpInstance.getName());
+
+
+                    try (PreparedStatement stmt = conn.prepareStatement(SQL_SELECT_KITS + SQL_AVOID_DELIVERED)) {
+                        stmt.setString(1, ddpInstance.getDdpInstanceId());
+                        stmt.setFetchSize(LOOKUP_CHUNK_SIZE);
+                        try (ResultSet rs = stmt.executeQuery()) {
+                            int count = 0;
+
+                            while (rs.next() && count < LOOKUP_CHUNK_SIZE) {
+                                count++;
+                                String kitLabel = rs.getString("kit.kit_label");
+                                logger.info(kitLabel);
+
+                                DdpKit kit = new DdpKit(
+                                        rs.getString(DBConstants.DSM_KIT_REQUEST_ID),
+                                        rs.getString("kit." + DBConstants.KIT_LABEL),
+                                        rs.getString("kit." + DBConstants.DSM_TRACKING_TO),
+                                        rs.getString("kit." + DBConstants.TRACKING_RETURN_ID),
+                                        rs.getString("kit." + DBConstants.ERROR),
+                                        rs.getString("kit." + DBConstants.MESSAGE),
+                                        rs.getString("kit." + DBConstants.DSM_RECEIVE_DATE),
+                                        rs.getString("req." + DBConstants.COLLABORATOR_PARTICIPANT_ID),
+                                        rs.getString("req." + DBConstants.EXTERNAL_ORDER_NUMBER),
+                                        rs.getBoolean("kit." + DBConstants.CE_ORDER),
+                                        rs.getString("kth." + DBConstants.KIT_SHIPPING_HISTORY),
+                                        rs.getString("kth." + DBConstants.KIT_RETURN_HISTORY),
+                                        rs.getString("req." + DBConstants.DDP_INSTANCE_ID),
+                                        rs.getString("kit." + DBConstants.UPS_TRACKING_STATUS),
+                                        rs.getString("kit." + DBConstants.UPS_TRACKING_DATE),
+                                        rs.getString("kit." + DBConstants.UPS_RETURN_STATUS),
+                                        rs.getString("kit." + DBConstants.UPS_RETURN_DATE)
+                                );
+                                subsetOfKits.add(kit);
+                                logger.info("added " + kitLabel);
+
+                            }
+                        }
+                    }
+                    conn.close();
+                }
+
+
+
+            }
+            String json = new Gson().toJson(subsetOfKits.toArray(), DdpKit[].class);
```

Comment:
Thinking back to some of the scale issues we saw earlier in GAE: serializing potentially large objects to Strings is something we should avoid.  Instead, use a more stream-oriented approach.  We should also avoid logging large objects like this on L112.

Consider passing `ArrayList<DdpKit>` instead of `json` to `kitTrackerPubSubPublisher.publishWithErrorHandlerExample`.  Then within that method, look for a more stream-oriented methods such as  `ByteString.readFrom()` and stream-oriented `Gson` methods.